src/main/java/io/confluent/connect/elasticsearch/DataConverter.java

```diff
@@ -200,11 +202,19 @@ static Object preProcessValue(Object value, Schema schema, Schema newSchema) {
         }
         return result;
       case MAP:
-        keySchema = schema.keySchema();
-        valueSchema = schema.valueSchema();
-        ArrayList<Struct> mapStructs = new ArrayList<>();
-        Map<?, ?> map = (Map<?, ?>) value;
+        Schema keySchema = schema.keySchema();
         Schema newValueSchema = newSchema.valueSchema();
+        Schema valueSchema = schema.valueSchema();
```

Comment:
so this one we can't actually do because of the fallthrough with the case statement. When I tried adding a `newKeySchema` outside the scope of the string keyed map, I get test failures because of problems looking up a key schema for non-map objects. It's a bit ugly, but I think it's the best we can do. `valueSchema` does end up getting used in the `STRUCT` case as well, so it has multiple uses further down. I flipped the order though because that was nonsense :)